diff --git a/node_modules/ganache-core/lib/subproviders/geth_api_double.js b/node_modules/ganache-core/lib/subproviders/geth_api_double.js
index f56afe6..56c6bed 100644
--- a/node_modules/ganache-core/lib/subproviders/geth_api_double.js
+++ b/node_modules/ganache-core/lib/subproviders/geth_api_double.js
@@ -164,6 +164,12 @@ GethApiDouble.prototype.eth_hashrate = function(callback) {
   callback(null, "0x0");
 };
 
+GethApiDouble.prototype.eth_chainId = function(callback) {
+  // ulfi
+  callback(null, "0x1");
+};
+
+
 GethApiDouble.prototype.eth_gasPrice = function(callback) {
   callback(null, utils.addHexPrefix(this.state.gasPrice()));
 };
