diff --git a/node_modules/web3-core-method/dist/web3-core-method.cjs.js b/node_modules/web3-core-method/dist/web3-core-method.cjs.js
index 13fcdbf..029780c 100644
--- a/node_modules/web3-core-method/dist/web3-core-method.cjs.js
+++ b/node_modules/web3-core-method/dist/web3-core-method.cjs.js
@@ -1026,6 +1026,10 @@ function (_AbstractMethod) {
             return;
           }
           _this2.promiEvent.emit('confirmation', confirmations, _this2.afterExecution(receipt));
+          // ulfi: quick hack to finish promise
+          _this2.promiEvent.removeAllListeners();
+          transactionConfirmationSubscription.unsubscribe();
+          _this2.promiEvent.resolve(_this2.afterExecution(receipt));
         }, function (error) {
           _this2.handleError(error, receipt, confirmations);
         }, function () {
diff --git a/node_modules/web3-core-method/dist/web3-core-method.esm.js b/node_modules/web3-core-method/dist/web3-core-method.esm.js
index 28204f8..2b2d3e8 100644
--- a/node_modules/web3-core-method/dist/web3-core-method.esm.js
+++ b/node_modules/web3-core-method/dist/web3-core-method.esm.js
@@ -628,6 +628,10 @@ class AbstractObservedTransactionMethod extends AbstractMethod {
           return;
         }
         this.promiEvent.emit('confirmation', confirmations, this.afterExecution(receipt));
+        // ulfi: quick hack to finish promise
+        this.promiEvent.removeAllListeners();
+        transactionConfirmationSubscription.unsubscribe();
+        this.promiEvent.resolve(this.afterExecution(receipt));
       }, error => {
         this.handleError(error, receipt, confirmations);
       }, () => {
diff --git a/node_modules/web3-core-method/dist/web3-core-method.umd.js b/node_modules/web3-core-method/dist/web3-core-method.umd.js
index 985e3f4..7fb79d1 100644
--- a/node_modules/web3-core-method/dist/web3-core-method.umd.js
+++ b/node_modules/web3-core-method/dist/web3-core-method.umd.js
@@ -1024,6 +1024,10 @@
                 return;
               }
               _this2.promiEvent.emit('confirmation', confirmations, _this2.afterExecution(receipt));
+              // ulfi: quick hack to finish promise
+              _this2.promiEvent.removeAllListeners();
+              transactionConfirmationSubscription.unsubscribe();
+              _this2.promiEvent.resolve(_this2.afterExecution(receipt));
             }, function (error) {
               _this2.handleError(error, receipt, confirmations);
             }, function () {
