diff --git a/node_modules/web3-eth-contract/dist/web3-eth-contract.cjs.js b/node_modules/web3-eth-contract/dist/web3-eth-contract.cjs.js
index 3b6752c..e59b97e 100644
--- a/node_modules/web3-eth-contract/dist/web3-eth-contract.cjs.js
+++ b/node_modules/web3-eth-contract/dist/web3-eth-contract.cjs.js
@@ -724,7 +724,8 @@ function (_CallMethod) {
       if (outputs.length > 1) {
         return this.abiCoder.decodeParameters(outputs, response);
       }
-      return this.abiCoder.decodeParameter(outputs[0].type, response);
+//      return this.abiCoder.decodeParameter(outputs[0].type, response);
+      return this.abiCoder.decodeParameter(outputs[0], response);
     }
   }]);
   return CallContractMethod;
diff --git a/node_modules/web3-eth-contract/dist/web3-eth-contract.esm.js b/node_modules/web3-eth-contract/dist/web3-eth-contract.esm.js
index 65f0c1b..ebb3c8d 100644
--- a/node_modules/web3-eth-contract/dist/web3-eth-contract.esm.js
+++ b/node_modules/web3-eth-contract/dist/web3-eth-contract.esm.js
@@ -563,7 +563,8 @@ class CallContractMethod extends CallMethod {
     if (outputs.length > 1) {
       return this.abiCoder.decodeParameters(outputs, response);
     }
-    return this.abiCoder.decodeParameter(outputs[0].type, response);
+//    return this.abiCoder.decodeParameter(outputs[0].type, response);
+    return this.abiCoder.decodeParameter(outputs[0], response);
   }
 }
 
diff --git a/node_modules/web3-eth-contract/dist/web3-eth-contract.umd.js b/node_modules/web3-eth-contract/dist/web3-eth-contract.umd.js
index 81ebd63..73ea2b9 100644
--- a/node_modules/web3-eth-contract/dist/web3-eth-contract.umd.js
+++ b/node_modules/web3-eth-contract/dist/web3-eth-contract.umd.js
@@ -718,7 +718,8 @@
           if (outputs.length > 1) {
             return this.abiCoder.decodeParameters(outputs, response);
           }
-          return this.abiCoder.decodeParameter(outputs[0].type, response);
+//          return this.abiCoder.decodeParameter(outputs[0].type, response);
+          return this.abiCoder.decodeParameter(outputs[0], response);
         }
       }]);
       return CallContractMethod;
