diff --git a/node_modules/web3-providers/dist/web3-providers.cjs.js b/node_modules/web3-providers/dist/web3-providers.cjs.js
index 04ba291..e815eea 100644
--- a/node_modules/web3-providers/dist/web3-providers.cjs.js
+++ b/node_modules/web3-providers/dist/web3-providers.cjs.js
@@ -404,17 +404,18 @@ function (_AbstractSocketProvid) {
   }, {
     key: "onError",
     value: function onError(event) {
-      if (event.code === 'ECONNREFUSED') {
-        this.reconnect();
-        return;
-      }
+//      if (event.code === 'ECONNREFUSED') {
+//        this.reconnect();
+//        return;
+//      }
       _get(_getPrototypeOf(WebsocketProvider.prototype), "onError", this).call(this, event);
     }
   }, {
     key: "onClose",
     value: function onClose(closeEvent) {
       if (closeEvent.code !== 1000 || closeEvent.wasClean === false) {
-        this.reconnect();
+//        this.reconnect();
+        _get(_getPrototypeOf(WebsocketProvider.prototype), "onError", this).call(this, undefined);
         return;
       }
       _get(_getPrototypeOf(WebsocketProvider.prototype), "onClose", this).call(this);
diff --git a/node_modules/web3-providers/dist/web3-providers.esm.js b/node_modules/web3-providers/dist/web3-providers.esm.js
index 6ff1359..c65cae8 100644
--- a/node_modules/web3-providers/dist/web3-providers.esm.js
+++ b/node_modules/web3-providers/dist/web3-providers.esm.js
@@ -245,15 +245,16 @@ class WebsocketProvider extends AbstractSocketProvider {
     super.onMessage(messageEvent.data);
   }
   onError(event) {
-    if (event.code === 'ECONNREFUSED') {
-      this.reconnect();
-      return;
-    }
+//    if (event.code === 'ECONNREFUSED') {
+//      this.reconnect();
+//      return;
+//    }
     super.onError(event);
   }
   onClose(closeEvent) {
     if (closeEvent.code !== 1000 || closeEvent.wasClean === false) {
-      this.reconnect();
+//      this.reconnect();
+        super.onError(undefined);
       return;
     }
     super.onClose();
diff --git a/node_modules/web3-providers/dist/web3-providers.umd.js b/node_modules/web3-providers/dist/web3-providers.umd.js
index ee747c6..fe90edd 100644
--- a/node_modules/web3-providers/dist/web3-providers.umd.js
+++ b/node_modules/web3-providers/dist/web3-providers.umd.js
@@ -402,17 +402,18 @@
       }, {
         key: "onError",
         value: function onError(event) {
-          if (event.code === 'ECONNREFUSED') {
-            this.reconnect();
-            return;
-          }
+//          if (event.code === 'ECONNREFUSED') {
+//            this.reconnect();
+//            return;
+//          }
           _get(_getPrototypeOf(WebsocketProvider.prototype), "onError", this).call(this, event);
         }
       }, {
         key: "onClose",
         value: function onClose(closeEvent) {
           if (closeEvent.code !== 1000 || closeEvent.wasClean === false) {
-            this.reconnect();
+//            this.reconnect();
+            _get(_getPrototypeOf(WebsocketProvider.prototype), "onError", this).call(this, undefined);
             return;
           }
           _get(_getPrototypeOf(WebsocketProvider.prototype), "onClose", this).call(this);
